File: 0080-remove-duplicates-from-sorted-array-ii/0080-remove-duplicates-from-sorted-array-ii.java
class Solution {
    public int removeDuplicates(int[] nums) {
        int newIdx = 0;
        for(int i=0; i<nums.length;){
            int temp = nums[i];
            nums[i] = 10001;
            i++;
            nums[newIdx] = temp;
            newIdx++;

            if(i < nums.length && temp == nums[i]){
                nums[i] = 10001;
                i++;
                nums[newIdx] = temp;
                newIdx++;
                if(i >= nums.length) {
                    return newIdx;
                }
                while(i < nums.length && temp == nums[i]) i++;
            }
        }
        return newIdx;
    }
}

File: 0080-remove-duplicates-from-sorted-array-ii/0080-remove-duplicates-from-sorted-array-ii.cpp
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        int tmp=1;
        for(int i=1; i<nums.size(); i++){
            if(nums[i]==nums[i-1]){
                tmp++;
                if(tmp>2) {
                nums.erase(nums.begin()+i);
                i--; tmp--;
                }
            }else tmp=1;
        }
        return nums.size();
    }    
};
